src/selectors/nodes.js

```diff
@@ -255,23 +266,32 @@ export const getNodeDataObject = createSelector(
     }, {})
 );
 
-/**
- * Return the modular pipelines tree with full data for each tree node for display.
- */
 export const getModularPipelinesTree = createSelector(
-  [(state) => state.modularPipeline.tree, getNodeDataObject],
-  (modularPipelinesTree, nodes) => {
+  [
+    (state) => state.modularPipeline.tree,
+    (state) => state.slice.apply,
+    getNodeDataObject,
+    getSlicedPipeline,
+  ],
+  (modularPipelinesTree, isSlicingPipelineApplied, nodes, slicedPipeline) => {
     if (!modularPipelinesTree) {
       return {};
     }
+
+    // Determine the relevant nodes based on whether slicing is applied
+    const relevantNodes = isSlicingPipelineApplied
+      ? filterNodesFromSlicingPipeline(nodes, slicedPipeline)
+      : nodes;
+
     for (const modularPipelineID in modularPipelinesTree) {
       modularPipelinesTree[modularPipelineID].data = {
         ...nodes[modularPipelineID],
```

Comment:
Shouldn't this have ...relevantNodes[modularPipelineID] ?

Reply:
hmm yeah I don't think it makes a different as at this level it is about modular pipeline, so nodes or relevant nodes will have the same data about modular pipeline

Reply:
but relavant nodes are subset of nodes right ? So the relevant nodes might not have the modularPipelineID ?

Reply:
hey relevant nodes will have modularPipelineID , as well as nodes. That's because when we do the slicing, we only filtering out nodes but not modular pipelines